Mathlib/Topology/Bases.lean

```diff
@@ -119,6 +120,39 @@ theorem isTopologicalBasis_of_subbasis {s : Set (Set α)} (hs : t = generateFrom
     exact ⟨{t}, ⟨finite_singleton t, singleton_subset_iff.2 ht⟩, rfl⟩
 #align topological_space.is_topological_basis_of_subbasis TopologicalSpace.isTopologicalBasis_of_subbasis
 
+theorem isTopologicalBasis_of_subbasis_of_finiteInter {s : Set (Set α)} (hsg : t = generateFrom s)
+    (hsi : FiniteInter s) : IsTopologicalBasis s := by
+  have e1 {s : Set (Set α)} (hsi : FiniteInter s) :
+    (fun g => ⋂₀ g) '' { f : Set (Set α) | f.Finite ∧ f ⊆ s } = s := by
+    rw [le_antisymm_iff]
+    constructor
+    · intro F
+      simp only [mem_image, mem_setOf_eq, forall_exists_index, and_imp]
+      intro G hG hGs hi
+      rw [← hi]
+      let G' := Finite.toFinset hG
+      have eG : G = G' := by
+         simp only [Finite.coe_toFinset, G']
+      rw [eG]
+      apply @FiniteInter.finiteInter_mem _ s hsi G'
+      rw [← eG]
+      exact hGs
+    · intro t hts
+      use {t}
+      simp only [mem_setOf_eq, finite_singleton, singleton_subset_iff, true_and, sInter_singleton,
+        and_true, hts]
+  rw [← e1 hsi]
+  exact isTopologicalBasis_of_subbasis hsg
+
+theorem isTopologicalBasis_of_subbasis_of_inter {r : Set (Set α)} (hsg : t = generateFrom r)
+    (hsi : ∀ ⦃s⦄, s ∈ r → ∀ ⦃t⦄, t ∈ r → s ∩ t ∈ r) : IsTopologicalBasis (insert univ r) := by
+  apply isTopologicalBasis_of_subbasis_of_finiteInter _ (FiniteInter.mk₂ hsi)
+  rw [hsg]
+  apply le_antisymm _ (generateFrom_anti <| subset_insert univ r)
+  exact le_generateFrom fun _ hs => (by
+    rcases hs with h | h; simp only [h, isOpen_univ];
+    exact isOpen_generateFrom_of_mem h)
```

Comment:
Just extract the lemma and the `simp` lemma here (of course, put it in the appropriate place):
```lean
lemma generateFrom_insert_of_generateOpen {α : Type*} {s : Set (Set α)} {t : Set α}
    (ht : GenerateOpen s t) : generateFrom (insert t s) = generateFrom s := by
  refine le_antisymm (generateFrom_anti <| subset_insert t s) (le_generateFrom ?_)
  rintro t (rfl | h)
  · exact ht
  · exact isOpen_generateFrom_of_mem h

@[simp]
lemma generateFrom_insert_univ {α : Type*} {s : Set (Set α)} :
    generateFrom (insert univ s) = generateFrom s :=
  generateFrom_insert_of_generateOpen .univ
```
And then this proof becomes:
```suggestion
    (hsi : ∀ ⦃s⦄, s ∈ r → ∀ ⦃t⦄, t ∈ r → s ∩ t ∈ r) : IsTopologicalBasis (insert univ r) :=
  isTopologicalBasis_of_subbasis_of_finiteInter (by simpa using hsg) (FiniteInter.mk₂ hsi)
```

Reply:
I assume the correct place for the lemmas is in `Topology/Order`?